modules/core/04-channel/keeper/upgrade_test.go

```diff
@@ -129,6 +130,245 @@ func (suite *KeeperTestSuite) TestChanUpgradeInit() {
+
+			err := path.EndpointA.ChanUpgradeInit()
+			suite.Require().NoError(err)
+
+			// commit a block to update chain A for correct proof querying
+			path.EndpointA.Chain.Coordinator.CommitBlock(path.EndpointA.Chain)
+			// update chainB's client of chain A to account for ChanUpgradeInit
+			suite.Require().NoError(path.EndpointB.UpdateClient())
+
+			proposedConnectionHops = []string{path.EndpointB.ConnectionID}
+			upgrade := types.NewUpgrade(
+				types.NewUpgradeFields(
+					types.UNORDERED, proposedConnectionHops, fmt.Sprintf("%s-v2", mock.Version),
+				),
+				types.NewTimeout(path.EndpointA.Chain.GetTimeoutHeight(), 0),
+				0,
+			)
+			counterpartyUpgrade = path.EndpointA.GetProposedUpgrade()
+			// expSequence = 1
+
+			tc.malleate()
+
+			// we need to update the clients again because malleation has changed the channel state
+			suite.Require().NoError(path.EndpointA.UpdateClient())
+			suite.Require().NoError(path.EndpointB.UpdateClient())
+
+			proofCounterpartyChannel, proofCounterpartyUpgrade, proofHeight := path.EndpointB.QueryChannelUpgradeProof()
+
+			_, err = suite.chainB.GetSimApp().IBCKeeper.ChannelKeeper.ChanUpgradeTry(
+				suite.chainB.GetContext(),
+				path.EndpointB.ChannelConfig.PortID,
+				path.EndpointB.ChannelID,
+				proposedConnectionHops,
+				upgrade.Timeout,
+				counterpartyUpgrade,
+				path.EndpointA.GetChannel().UpgradeSequence,
+				proofCounterpartyChannel,
+				proofCounterpartyUpgrade,
+				proofHeight)
+
+			if tc.expPass {
+				suite.Require().NoError(err)
+				// suite.Require().Equal(expSequence, path.EndpointB.GetChannel().UpgradeSequence)
+				// suite.Require().Equal(mock.Version, path.EndpointB.GetChannel().Version)
+				// suite.Require().Equal(path.EndpointB.GetChannel().State, types.TRYUPGRADE)
+			} else {
+				suite.Require().Error(err)
+			}
+		})
+	}
+}
+
+func (suite *KeeperTestSuite) TestChanUpgradeTry_CrossingHellos() {
+	var (
+		path *ibctesting.Path
+		// expSequence                 uint64
+		upgrade                     types.Upgrade
+		counterpartyUpgrade         types.Upgrade
+		counterpartyUpgradeSequence uint64
+	)
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		expPass  bool
+	}{
+		{
+			"success",
+			func() {},
+			true,
+		},
+		// {
+		// 	"success: counterparty sequence > channel.UpgradeSequence",
+		// 	func() {
+		// 		counterpartyUpgradeSequence = 5
+		// 		expSequence = 5
+		// 	},
+		// 	true,
+		// },
+		// {
+		// 	"fail: upgrade fields have changed",
+		// 	func() {
+		// 		counterpartyUpgrade.Fields.Ordering = types.ORDERED
+		// 		counterpartyUpgrade.Fields.Version = fmt.Sprintf("%s-v3", mock.Version)
+		// 	},
+		// 	false,
+		// },
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		suite.Run(tc.name, func() {
+			suite.SetupTest()
+
+			path = ibctesting.NewPath(suite.chainA, suite.chainB)
+			suite.coordinator.Setup(path)
+
+			// chainA UpgradeInit
```

Comment:
```suggestion
```
nit: the comment isn't including more information than the call to the functino `path.EndpointA.ChanUpgradeInit`